Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1344,6 +1344,7 @@ class BrowserViewController: UIViewController,
         }
         urlBar.currentURL = tab.url?.displayURL
         urlBar.locationView.tabDidChangeContentBlocking(tab)
+        urlBar.locationView.updateShoppingCartButtonVisibility(for: tab)
```

Comment:
Just trying to understand, is there any feature flag for this feature? Shouldn't this be behind it?

Reply:
this line checks the flag:  shoppingCartButton.isHidden = !product.isShoppingCartButtonVisible || tab.isPrivate
the check is in the ShoppingProduct class: 
var isShoppingCartButtonVisible: Bool {
        return product != nil && isFakespotFeatureEnabled
}

Reply:
Ok great 😄 Thank for the explanation